src/Symfony/Component/Serializer/Instantiator/Instantiator.php

```diff
@@ -116,4 +239,32 @@ private function denormalizeParameter(\ReflectionClass $class, \ReflectionParame
 
         return $parameterData;
     }
+
+    protected function getConstructor(\ReflectionClass $reflectionClass): ?\ReflectionMethod
+    {
+        return $reflectionClass->getConstructor();
```

Comment:
What is the benefit of not calling the method directly?

Reply:
Sometimes you want to use another method (static) or a factory to create an object, this method can be overloaded for this uses.

You can check https://github.com/symfony/symfony/pull/30956/files#diff-3c786d6cd63848e8aea581f9d2e573adR19-R33 for a working exemple.

Reply:
Couldn't the context hold information what other method to call?

That means that we can also declare this class as `final` and we dont need to use inheritance like this.